src/module_handle.cc

```diff
@@ -0,0 +1,279 @@
+		"getModuleRequest", Parameterize<decltype(&ModuleHandle::GetModuleRequest<1>), &ModuleHandle::GetModuleRequest<1>>(),
+		"getModuleRequestSync", Parameterize<decltype(&ModuleHandle::GetModuleRequest<0>), &ModuleHandle::GetModuleRequest<0>>(),
+		"setDependency", Parameterize<decltype(&ModuleHandle::SetDependency), &ModuleHandle::SetDependency>(),
+		"release", Parameterize<decltype(&ModuleHandle::Release), &ModuleHandle::Release>(),
+		"instantiate", Parameterize<decltype(&ModuleHandle::Instantiate<1>), &ModuleHandle::Instantiate<1>>(),
+		"instantiateSync", Parameterize<decltype(&ModuleHandle::Instantiate<0>), &ModuleHandle::Instantiate<0>>(),
+		"evaluate", Parameterize<decltype(&ModuleHandle::Evaluate<1>), &ModuleHandle::Evaluate<1>>(),
+		"evaluateSync", Parameterize<decltype(&ModuleHandle::Evaluate<0>), &ModuleHandle::Evaluate<0>>(),
+		"getModuleNamespace", Parameterize<decltype(&ModuleHandle::GetModuleNamespace<1>), &ModuleHandle::GetModuleNamespace<1>>(),
+		"getModuleNamespaceSync", Parameterize<decltype(&ModuleHandle::GetModuleNamespace<0>), &ModuleHandle::GetModuleNamespace<0>>()
+	));
+}
+
+std::unique_ptr<Transferable> ModuleHandle::TransferOut() {
+	return std::make_unique<ModuleHandleTransferable>(isolate, _module);
+}
+
+
+
+struct GetModuleRequestRunner : public ThreePhaseTask {
+	std::unique_ptr<Transferable> result;
+	std::shared_ptr<RemoteHandle<v8::Module>> _module;
+	std::size_t _index;
+
+	GetModuleRequestRunner(std::shared_ptr<RemoteHandle<v8::Module>> myModule, std::size_t index)
+	  : _module(std::move(myModule)), _index(index)
+	{
+	}
+
+	void Phase2() final {
+		auto request = this->_module->Deref()->GetModuleRequest(this->_index);
+		result = ExternalCopy::CopyIfPrimitive(request);
+	}
+
+	Local<Value> Phase3() final {
+		if (result) {
+			return result->TransferIn();
+		} else {
+			return v8::Undefined(v8::Isolate::GetCurrent()).As<v8::Value>();
+		}
+	}
+
+};
+
+
+v8::Local<v8::Value> ModuleHandle::GetModuleRequestsLength() {
+  const std::size_t length = this->_module->Deref()->GetModuleRequestsLength();
+  Isolate* isolate = Isolate::GetCurrent();
+  return v8::Integer::New(isolate, length);
+}
+
+template <int async>
+v8::Local<v8::Value> ModuleHandle::GetModuleRequest(v8::Local<v8::Value> index) {
+	return ThreePhaseTask::Run<async, GetModuleRequestRunner>(*this->isolate, this->_module, index->Int32Value());
+}
+
+
+v8::Local<v8::Value> ModuleHandle::SetDependency(v8::Local<v8::Value> value, ModuleHandle* module_handle_ptr) {
+	if (!value->IsString()) {
+		return v8::Boolean::New(v8::Isolate::GetCurrent(), false);
+	}
+	v8::Local<v8::String> key = value->ToString();
+	v8::String::Utf8Value keyAsCharPointer(v8::Isolate::GetCurrent(), key);
+	if (!keyAsCharPointer.length()) {
+		// UTF8-conversion error probably
+		return v8::Boolean::New(v8::Isolate::GetCurrent(), false);
+	}
+	// Assume the code cannot invoke SetDependency() in parallel, otherwise will the code below not be thread safe
```

Comment:
This is where a mutex comes in handy, so you don't have to make this assumption.